src/backend/document.cpp

```diff
@@ -239,7 +246,7 @@ std::vector<Glib::RefPtr<Page>> Document::loadPages(const Document::FileData& fi
     const Glib::ustring basename = Glib::filename_display_basename(fileData.originalFile->get_path());
     std::vector<Glib::RefPtr<Page>> result;
 
-    for (int i = 0; i < fileData.popplerDocument->pages(); ++i) {
+    for (unsigned int i = 0; i < fileData.popplerDocument->pages(); ++i) {
```

Comment:
The `i` variable should be a plain `int`, since that is the datatype returned by `fileData.popplerDocument->pages()`.